plugin-server/tests/utils/concurrencyController.test.ts

```diff
@@ -0,0 +1,153 @@
+import { ConcurrencyController } from '../../src/utils/concurrencyController'
```

Comment:
these tests are just a straight-forward copy of the tests in the front-end codebase